src/Atmos/Model/moisture.jl

```diff
@@ -22,7 +22,7 @@ function internal_energy(m::MoistureModel, state::Vars, aux::Vars)
   q_pt = get_phase_partition(m, state)
   ρinv = 1 / state.ρ
   ρe_kin = ρinv*sum(abs2, state.ρu)/2
-  ρe_pot = state.ρ * grav * aux.coord.z
+  ρe_pot = state.ρ * grav * aux.coord[3]
```

Comment:
Note that `aux.coord[3]` is not the radial direction.  This code needs to change to work on the sphere.

Reply:
Yes: we need a way that will work for both the rectangle and sphere

Reply:
My recommendation would be that we store `ϕ` and `∇ϕ`. We can then check in one place how to set `ϕ` (probably in `init_aux` if the grid type is sphere)

Reply:
Agreed, that does seem like the simplest solution for now.